lib/src/over_react_test/common_component_util.dart

```diff
@@ -502,18 +501,65 @@ void testRequiredProps(BuilderOnlyUiFactory factory, dynamic childrenFactory(),
   }
 
   test('nullable props', () {
-    nullableProps.forEach((String propKey) {
-      var badRenderer = () => render((factory()..remove(propKey)(childrenFactory())));
 
-      expect(badRenderer, throwsPropError_Required(propKey, keyToErrorMessage[propKey]), reason: 'should throw when the required, nullable prop $propKey is not set');
+    if (!isComponent2) {
+      nullableProps.forEach((String propKey) {
+        var badRenderer = () => render((factory()..remove(propKey))(childrenFactory()));
 
-      var propsToAdd = {propKey: null};
-      badRenderer = () => render((factory()
-        ..addAll(propsToAdd)
-      )(childrenFactory()));
+        expect(badRenderer, throwsPropError_Required(propKey, keyToErrorMessage[propKey]), reason: 'should throw when the required, nullable prop $propKey is not set');
 
-      expect(badRenderer, returnsNormally, reason: 'does not throw when the required, nullable prop $propKey is set to null');
-    });
+        var propsToAdd = {propKey: null};
+        badRenderer = () => render((factory()
+          ..addAll(propsToAdd)
+        )(childrenFactory()));
+
+        expect(badRenderer, returnsNormally, reason: 'does not throw when the required, nullable prop $propKey is set to null');
+      });
+    } else if (isComponent2) {
+      PropTypes.resetWarningCache();
+
+      List<String> consoleErrors = [];
+      JsFunction originalConsoleError = context['console']['error'];
+      context['console']['error'] = new JsFunction.withThis((self, [message, arg1, arg2, arg3,  arg4, arg5]) {
+        consoleErrors.add(message);
+        originalConsoleError.apply([message, arg1, arg2, arg3,  arg4, arg5],
+            thisArg: self);
+      });
+
+      final reactComponentFactory = factory().componentFactory as
+      ReactDartComponentFactoryProxy2; // ignore: avoid_as
+
+      nullableProps.forEach((String propKey) {
+        if (!reactComponentFactory.defaultProps.containsKey(propKey)) {
+
+          wrapInTryCatch(() => mount((factory()
+              ..remove(propKey)
+            )(childrenFactory())));
+
+          expect(consoleErrors, isNotEmpty, reason: 'should have outputted a warning');
+
+          if (keyToErrorMessage[propKey] != '') {
+            expect(consoleErrors, [contains(keyToErrorMessage[propKey])], reason: '$propKey is not set');
+          }
+
+          consoleErrors = [];
+          PropTypes.resetWarningCache();
+        }
+
+        var propsToAdd = {propKey: null};
+
+        wrapInTryCatch(() => mount((factory()
+            ..addAll(propsToAdd)
+          )(childrenFactory())));
+
+        expect(consoleErrors, isEmpty, reason: 'should not have output a warning');
+
+        consoleErrors = [];
+        PropTypes.resetWarningCache();
+      });
+
+      addTearDown(() => context['console']['error'] = originalConsoleError);
```

Comment:
This is exactly how I did it here and in the WSD proptype test utils, but Greg pointed out yesterday / today that the `addTearDown` should be put right under `originalConsoleError` to make sure it has a chance get added in the case that tests go badly. If you also wanted to fix my mistake on line 499, that would be very nice of you 😃

Reply:
lol that makes perfect sense damn